spec.html

```diff
@@ -13723,14 +13723,17 @@ <h1>
-    <p>Built-in function objects that are not identified as constructors do not implement the [[Construct]] internal method unless otherwise specified in the description of a particular function. When a built-in constructor is called as part of a `new` expression the _argumentsList_ parameter of the invoked [[Construct]] internal method provides the values for the built-in constructor's named parameters.</p>
-    <p>Built-in functions that are not constructors do not have a *"prototype"* property unless otherwise specified in the description of a particular function.</p>
-    <p>If a built-in function object is not implemented as an ECMAScript function it must provide [[Call]] and [[Construct]] internal methods that conform to the following definitions:</p>
+    <p>A built-in function object is an ordinary object; it must satisfy the requirements for ordinary objects set out in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.</p>
+    <p>In addition to the internal slots required of every ordinary object, a built-in function object must also have the following internal slots:</p>
+    <ul>
+      <li>[[Realm]], a Realm Record that represents the realm in which the function was created.</li>
+      <li>[[InitialName]], a String that is the initial name of the function. It is used by <emu-xref href="#sec-function.prototype.tostring"></emu-xref>.</li>
+    </ul>
+    <p>The initial value of a built-in function object's [[Prototype]] internal slot is %Function.prototype%, unless otherwise specified.</p>
+    <p>A built-in function object must have a [[Call]] internal method that conforms to the definition in <emu-xref href="#sec-built-in-function-objects-call-thisargument-argumentslist"></emu-xref>.</p>
+    <p>A built-in function object has a [[Construct]] internal method if and only if it is described as a “constructor”, or some algorithm in this specification explicitly sets its [[Construct]] internal method. Such a [[Construct]] internal method must conform to the definition in <emu-xref href="#sec-built-in-function-objects-construct-argumentslist-newtarget"></emu-xref>.</p>
+    <p>For a built-in function object defined in this specification, the behaviour specified for it via algorithm steps or other means is “the specification of _F_” for the purposes of step <emu-xref href="#step-call-builtin-function-result"></emu-xref> of both <emu-xref href="#sec-built-in-function-objects-call-thisargument-argumentslist" title></emu-xref> and (if applicable) <emu-xref href="#sec-built-in-function-objects-construct-argumentslist-newtarget" title></emu-xref>.</p>
```

Comment:
I don't like using the one step reference for both algorithms here. In the long run, I'd prefer not to define `[[Construct]]` as a modification to `[[Call]]`, but for now I would prefer to add a step reference to the replacement step and reference that.

```suggestion
    <p>For a built-in function object defined in this specification, the behaviour specified for it via algorithm steps or other means is “the specification of _F_” for the purposes of step <emu-xref href="#step-call-builtin-function-result"></emu-xref> of <emu-xref href="#sec-built-in-function-objects-call-thisargument-argumentslist" title></emu-xref> and (if applicable) step <emu-xref href="#step-construct-builtin-function-result"></emu-xref> of <emu-xref href="#sec-built-in-function-objects-construct-argumentslist-newtarget" title></emu-xref>.</p>
```

Reply:
> In the long run, I'd prefer not to define `[[Construct]]` as a modification to `[[Call]]`, but for now ...

Why "in the long run"? PR #2637 is ready to go. If that gets merged, then I can change the text here to just "step 10 of BuiltinCallOrConstruct".

I'm fine with having this PR wait for #2637.

Reply:
I am also fine with ordering this behind #2637.

Reply:
#2637 has been merged. You can now make the requested change.

Reply:
Done.